src/types.ts

```diff
@@ -28,32 +28,26 @@ export type SchemaType =
   | "[]string"
   | "[]int"
   | "[]uint"
-  | "[]uuid";
+  | "[]uuid"
+  | `[${number}]${"f16" | "f32"}`;
 
-export type Schema =
-  | {
-      [K in string]: {
-        type?: SchemaType;
-        filterable?: boolean;
-        /**
-         * @deprecated use `full_text_search` instead
-         */
-        bm25?: boolean | Partial<FTSParams>;
-        full_text_search?: boolean | Partial<FTSParams>;
-      };
-    }
-  | {
-      /**
-       * Whether the upserted vectors are of type f16 or f32.
-       *
-       * To use f16 vectors, this field needs to be explicitly specified in
-       * the schema when first creating (i.e. upserting to) a namespace.
-       */
-      vector?: {
-        type: `[${number}]${"f16" | "f32"}`;
-        ann: boolean;
-      };
-    };
+export type Schema = Record<
+  string,
+  {
+    type?: SchemaType;
+    filterable?: boolean;
+    /**
+     * @deprecated use `full_text_search` instead
+     */
+    bm25?: boolean | Partial<FTSParams>;
+    full_text_search?: boolean | Partial<FTSParams>;
+    /**
+     * Used with the `type` field to specify the upserted vector float type.
+     * See https://turbopuffer.com/docs/schema#param-vector.
+     */
```

Comment:
this comment is a bit off - the purpose of this field is to indicate whether a vector should be indexed with an ANN index or not

Reply:
my intention was to indicate that you currently cannot use it by itself and need to use it with type, which together allows you to set the float type, etc. but i just made it weird. removed that line, better this way. There's always the documentation on the website